### PR TITLE
Fix #3179 Update font awesome

### LINKED ISSUE #3179: Update FontAwesome to FontAwesome 5
### Describe the problem

FontAwesome 4.7 (a version made back in 2016) is no longer being maintained and needs to be replaced with FontAwesome 5. This will be a very annoying process as basically all the icons and classes have changed.

### Describe the solution you'd like

- [ ] First, update FontAwesome (This will break all icons more than likely)
- [ ] Fully replace all icons with the new classes

### PR DESCRIPTION
**Pull request by stu01509**
Fixes #3179 

#### Short description of what this resolves:
Update the npm package version.
Modify fontAwesome config.yaml.

### CODE REVIEW

**Comment by stu01509** [comment]
Hi @bradymiller and @tywrenn 

I had a little bit confused in FontAwesome.scss file src: url("/public/assets/@fortawesome/fontawesome-free/webfonts") format("truetype");

I try to redirect a new path, But font-awesome still broke the same,
I also open dev-tools to debug, I found the fontawesome-webfont.ttf was not found.

image

**Comment by tywrenn** [comment]
@stu01509 I'll give you instructions when I get home on how to proceed with changing this as this part is the most complex part

**Comment by stu01509** [comment]
@tywrenn Ok, Thanks for your help.

**Comment by tywrenn** [comment]
@stu01509 So the proper filename isn't `font-awesome.min.css` anymore. It's now `all.min.css`.

**Comment by stu01509** [comment]
Hi @tywrenn 

Thanks for your instructions, Now font-awesome successful to upgrade font-awesome 5,
But I have a question, Do I need to upgrade``` FontAwesome.scss``` font-face src path?

**Comment by tywrenn** [comment]
@stu01509 It should be the following:
`src: url("/public/assets/@fortawesome/fontawesome-free/webfonts/fa-regular-400.ttf") format("truetype");`

For some reason it's erroring still and i'm not sure why.

**Comment by tywrenn** [comment]
Crap just realized it was select2 SCSS

**Comment by stu01509** [comment]
Hi @tywrenn 

I follow your src path, but FontAwesome.scss CSS style icon still broken.

I modified these.
```scss
@font-face {
    font-family: "Font Awesome 5 Free";
    src: url("/public/assets/@fortawesome/fontawesome-free/webfonts/fa-regular-400.ttf") format("truetype");
}

.btn::before {
    font-family: "Font Awesome 5 Free";
    display: inline;
}
```

**Comment by tywrenn** [comment]
@bradymiller Docker broke us here again. I changed the FontAwesome.scss file and it isn't changing when it compiles again.

**Comment by bradymiller** [comment]
ensure follow step 5 here on the docker:
https://github.com/openemr/openemr/blob/master/CONTRIBUTING.md

I am guessing that this is a cache issue on the browser

**Comment by tywrenn** [comment]
@stu01509 Do what @bradymiller said as that worked for me.

**Comment by stu01509** [comment]
Hi @tywrenn 

I adjust this then almost icon looks like good.

```scss
.btn::before {
    font-family: "Font Awesome 5 Free";
    display: inline;
    font-weight: 900;
}
```

**Comment by stu01509** [comment]
Hi @bradymiller and @tywrenn 

Should I fix the broken icon on this PR or create another PR to fix it?

**Comment by tywrenn** [comment]
@stu01509 Disregard the FontAwesome.scss file for now --- I figured out it has more issues that need solving. Fixing the broken icons are also gonna be included with this pull since I planned to take it on in one shot.

**Comment by stu01509** [comment]
Hi @tywrenn 
OK, I just update ```config.yaml``` file.

**Comment by tywrenn** [comment]
@bradymiller @stu01509 Due to code flat out breaking during testing, I have decided that using the BTN SCSS is not gonna fly with FontAwesome 5. I'll be giving instructions on how to replace it as I figure out an alternative. The way they redid FontAwesome made it more difficult to keep working as I tried reviving it but to no avail.

**Comment by tywrenn** [comment]
@stu01509 @bradymiller Never mind -- found a solution. Copy and paste my code from here: https://codepen.io/tylerhyperHD/pen/vYOaMaL

**Comment by bradymiller** [comment]
hi @tywrenn , Can this then go into the codebase since this takes care of buttons? (Then can do a "sprint" to fix other icons on subsequent PRs.)

File: package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "replace-in-file": "^5.0.2"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-free": "^5.12.1",
```
Comment (bradymiller) [inline package.json]: we pin the versions in this section, so should just use 5.12.1

**Comment by tywrenn** [comment]
If we add the code I just sent we should be able to sprint and fix the rest. I already see a lot that need fixing

**Comment by stu01509** [comment]
Hi @bradymiller  @tywrenn 

I follow your code and paste to ```fontAwesome.scss``` it seems almost done,
Just some icon e.g., refresh button need to fix.

**Comment by tywrenn** [comment]
@stu01509 We actually don't need the font face in the file since it's already referenced in the framework which just means you have to have fontawesome put in the file ur using

**Comment by stu01509** [comment]
@tywrenn 

Ok, I removed font face and todo comment.

**Comment by bradymiller** [comment]
hi @stu01509 and @tywrenn , Just tried this and looking good on testing. Gonna bring in this PR and let the sprint fix begin :)
Thanks for the improvement!

**Pull request merged**